### PR TITLE
153 frontend privacy policy screen

### LINKED ISSUE #153: Frontend -- Privacy policy screen

### PR DESCRIPTION
**Pull request by bigboidanwithacan**
# What type of PR is this? (Check all that apply)

- [x] ✨**feature**: Introduces completely new code or new features.
- [ ] 🐛**fix**: Implements changes that fix a bug. Ideally, reference an issue if present.
- [ ] ♻️**refactor**: Includes any code-related change that is neither a fix nor a feature.
- [ ] ✅**build**: Encompasses all changes related to the build of the software, including changes to dependencies or the addition of new ones.
- [x] ⚡️**test**: Pertains to all changes regarding tests, whether adding new tests or modifying existing ones.
- [ ] 🚰**ci**: Involves all changes related to the configuration of continuous integration, such as GitHub Actions or other CI systems.
- [ ] 📚**docs**: Includes all changes to documentation, such as README files, or any other documentation present in the repository.
- [ ] 🗑️**chore**: Captures all changes to the repository that do not fit into the above categories.

# Description

What did you change? How did you change it?
Closes #153 
Created a privacy policy page. The policy was generated using ChatGPT.
https://chat.openai.com/share/edbfaba8-f225-4168-8d86-c8263ed9ddc5 

# Tests

How was this tested?

Created a snapshot using Jest.

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [ ] Manual tests
- [ ] Tests were NOT needed
- [ ] Other (explain below)

# [Optional] Screenshots

https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/38168108/9cdbe3e3-1c1d-42ac-a31b-c2470bcd6bb5

Initial Design
image

# Documentation

- [ ] Added to README.me
- [ ] Seperate document
- [x] NO documentation needed

Link to external documentation:

Link to ChatGPT prompt.
[https://chat.openai.com/share/edbfaba8-f225-4168-8d86-c8263ed9ddc5](url)

# [Optional] Are there any post-deployment tasks we need to perform?

### CODE REVIEW

**Review by StefanoRubini — COMMENTED** [review]
Hello Daniel. Thank you for taking the time to create this privacy policy screen. I appreciate how concise the policy is and the extent of which we collect data from our app users! 😃

File: .vscode/settings.json [hunk 1]
```diff
@@ -5,5 +5,7 @@
     }
   ],
   "typescript.tsdk": "node_modules\\typescript\\lib",
-  "discord.enabled": true
-}
\ No newline at end of file
+  "discord.enabled": true,
+  "editor.formatOnPaste": true,
```
Comment (thienguen) [inline .vscode/settings.json:9]: this could be delete but its alright

**Comment by thienguen** [comment]
1. move it to the setting page, privacy policy folder (create one) under (dashboard)\

image

2. TBD (we need taylor default nav to be merge, later)
3. bold privacy policy, put text in a box view (refer to the account setting page)
4. have a last updated section
5. use the label component Label.tsx in UI folder for the privacy policy

image

**Comment by thienguen** [comment]
reviewing now

File: apps/expo/src/app/(dashboard)/account-settings.tsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ const otherItems: NavigationListItem[] = [
   {
     title: 'Privacy Policy',
     iconName: 'shield-alt',
-    onPress: () => router.push('/'),
+    onPress: () => router.push('/privacy-policy' as Route<string>),
```
Comment (thienguen) [inline apps/expo/src/app/(dashboard)/account-settings.tsx:73]: image

change to this so you won't need that

File: apps/expo/src/app/privacy-policy/index.tsx [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+          >
+            {'\n'}Effective Date: [4/19/2024]{'\n'}
+          </Text>
+          <View className={tailwindClasses.mainTile}>
+            <Text className='text-2xl text-slate-200' style={{ paddingHorizontal: 10 }}>
+              Welcome to Barbell, your ultimate companion for fitness and workout routines. At
+              Barbell, we are committed to protecting your privacy and ensuring the security of your
+              personal information. This Privacy Policy outlines how we collect, use, disclose, and
+              protect your information when you use our mobile application ("App") and services.
+              {'\n\n'}
```
Comment (thienguen) [inline apps/expo/src/app/privacy-policy/index.tsx:60]: image

remove the \n\n at the end here, too many spaces comparatively

image

File: apps/expo/src/app/privacy-policy/index.tsx [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+              collect information from Spotify's API, including your current listening activity.
+              {'\n\n'}b. Workout Data:{'\n\t\t\t'} Barbell collects information about your workout
+              activities, including exercise routines, duration, intensity, and progress. This data
+              may be shared with friends and the public by default within the App.{'\n\n'}c. Device
+              Information:
+              {'\n\t\t\t'} We may collect information about your device, including its unique device
+              identifier, operating system, and mobile network information.{'\n\n'}d. Location
+              Information:{'\n\t\t\t'} With your consent, we may collect and process information
+              about your precise or approximate location when you use certain features of the App,
+              such as mapping your running routes.{'\n\n'}
```
Comment (thienguen) [inline apps/expo/src/app/privacy-policy/index.tsx:85]: image

update the styles as I updated here

File: apps/expo/src/app/privacy-policy/index.tsx [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+              workout data may be shared with friends and the public within the App. However, you
+              have control over the visibility settings of your profile and can adjust the privacy
+              settings accordingly.
+              {'\n\n'}c. Integration with Spotify:{'\n\t\t\t'} Integration with Spotify: If you
+              choose to link your Spotify account, we may access your Spotify data through Spotify's
+              API to display your current listening activity to others on the App. This feature
+              enhances social interaction within the fitness community, and you may choose to opt
+              out at any time from user settings should you change your mind.{'\n\n'}d. Analytics:
+              {'\n\t\t\t'} We may use aggregated and anonymized data for analytical purposes to
+              improve the performance and functionality of the App.{'\n\n'}
```
Comment (thienguen) [inline apps/expo/src/app/privacy-policy/index.tsx:109]: brother, why you have so many \n\n.

IMO, remove them

File: apps/expo/src/app/privacy-policy/index.tsx [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+              fontWeight: 'bold',
+            }}
+          >
+            Privacy Policy
+          </Text>
+
+          <View style={{ paddingHorizontal: 30, flexDirection: 'row' }}>
+            <Label text='Legal Document' textColor='#CFCFCF' backgroundColor='#48476D' />
+            <Text> </Text>
+            <Label text='Human-Friendly' textColor='#CFCFCF' backgroundColor='#48476D' />
```
Comment (thienguen) [inline apps/expo/src/app/privacy-policy/index.tsx:45]: have it something else than `human-friendly`.

and `legal documents`, sounds a bit erm...We might have something else?

File: apps/expo/src/app/privacy-policy/index.tsx [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+              website or through other communication channels.{'\n\n'}
+            </Text>
+          </View>
+
+          <View className={tailwindClasses.mainTile}>
+            <Text
+              className='text-2xl text-slate-200'
+              style={{ fontSize: 25, paddingHorizontal: 10 }}
+            >
+              8. Contact Us
```
Comment (thienguen) [inline apps/expo/src/app/privacy-policy/index.tsx:199]: remove this section, we have a dedicated contact us page #229

Comment (thienguen) [reply apps/expo/src/app/privacy-policy/index.tsx:199]: image

this is after I changed some styling 

compare to 

image

**Review by thienguen — CHANGES_REQUESTED** [review]

**Review by supakeshi — APPROVED** [review]
Initial styling looks clean and well segregated:

image

Scroll function works good and was able to go through the full document:

image

LGTM!

File: .vscode/settings.json (hunk 1 above)
Comment (bigboidanwithacan) [reply .vscode/settings.json]: Has been fixed

Comment (thienguen) [reply .vscode/settings.json]: sound good

**Review by BitsyBirb — APPROVED** [review]
I have nothing to say that T hasn't already, but in terms of deployment everything ran fine on my end. Good work; I'll take another quick gander once you get around to T's suggestions.

File: apps/expo/src/app/privacy-policy/index.tsx [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+const tailwindClasses = {
+  mainTile: 'm-4 p-4 rounded-lg bg-bb-dark-gray',
+  mainTileTitle: 'text-2xl text-slate-200',
+  mainTileItem: 'flex-row',
+  navigationListItemLabel: 'flex-auto m-2 self-center text-slate-200',
+  navigationListItemChevron: 'self-center text-slate-200',
+}
+
+const PrivacyPolicy = () => {
+  const [isEnabled, setIsEnabled] = useState(false)
```
[the remark this answers is not in the document]
Comment (bigboidanwithacan) [reply apps/expo/src/app/privacy-policy/index.tsx:17]: Thanks for that, I forgot to delete it after removing the feature that needed it.

File: apps/expo/src/app/(dashboard)/account-settings.tsx (hunk 2 above)
Comment (bigboidanwithacan) [reply apps/expo/src/app/(dashboard)/account-settings.tsx:73]: Thanks I didn't know you can get around it that way.

File: apps/expo/src/app/privacy-policy/index.tsx (hunk 3 above)
Comment (bigboidanwithacan) [reply apps/expo/src/app/privacy-policy/index.tsx:60]: Done, good style correction.

File: apps/expo/src/app/privacy-policy/index.tsx (hunk 4 above)
Comment (bigboidanwithacan) [reply apps/expo/src/app/privacy-policy/index.tsx:85]: Thank you this was exactly what i needed.

File: apps/expo/src/app/privacy-policy/index.tsx (hunk 5 above)
Comment (bigboidanwithacan) [reply apps/expo/src/app/privacy-policy/index.tsx:109]: You're right, I removed the '\n\n' at the end of all the sections

File: apps/expo/src/app/privacy-policy/index.tsx (hunk 6 above)
Comment (bigboidanwithacan) [reply apps/expo/src/app/privacy-policy/index.tsx:45]: Changed to `Data Collection` and `Third-Party Integration`

File: apps/expo/src/app/privacy-policy/index.tsx (hunk 7 above)
Comment (bigboidanwithacan) [reply apps/expo/src/app/privacy-policy/index.tsx:199]: Done

**Review by thienguen — APPROVED** [review]
Thanks for addressing all the changes I suggested, fantastic work 🙏

**Pull request merged**